changelog/_unreleased/2025-04-04-fix-product-media-import-with-whitespace-in-url.md

```diff
@@ -0,0 +1,14 @@
+---
+title: fix product media import with whitespace in url
+issue: 5868
+author: p.dinkhoff
+author_email: p.dinkhoff@shopware.com
+author_github: @p.dinkhoff
+---
+# Core
+* Changed: URLs generated by the `MediaUrlGenerator` will properly encode the file path to produce valid URLs in version `v6.8.0.0` (e.g. replace whitespaces as `%20`). The current implementation relies on the caller to do proper URL encoding with the returned string.
+___
+# Next Major Version Changes
+## Changed URL generation of `MediaUrlGenerator` to properly encode the file path to produce valid URLs
+* For example media files with spaces in their name now should be properly URL-encoded with `%20` by default, without doing URL-encoding only with the return value of the `MediaUrlGenerator`. Make sure to remove extra URL-encoding (e.g. usage of twig filter `encodeUrl`) on media entities to not accidentally double encode the URLs.
+* Changed twig filter `encodeMediaUrl` in `Storefront/Framework/Twig/Extension/UrlEncodingTwigFilter.php` will now return the URL in its already encoded form and is basically the same as `$media->getUrl()` with some extra checks.
```

Comment:
I think (please correct me if I'm wrong, @mitelg) this is not necessary anymore, since you aleady updated `UPGRADE-6.8.md`